opt/scopeserver/dataserver/utils/loom_file_handler.py

```diff
@@ -80,71 +56,51 @@ def change_loom_mode(
         return self.global_looms
 
     def set_global_data(self) -> None:
         self.global_looms = [x for x in os.listdir(self.loom_dir) if not os.path.isdir(os.path.join(self.loom_dir, x))]
 
-    def get_loom_connection(self, loom_file_path: str, mode: str = "r"):
+    def get_loom_connection(self, loom_file_path: Path, mode: str = "r") -> LoomConnection:
         logger.debug(f"Getting connection for {loom_file_path} in mode {mode}")
         return self.get_loom(loom_file_path=loom_file_path, mode=mode).get_connection()
 
-    def get_loom(self, loom_file_path: str, mode: str = "r"):
+    def get_loom(self, loom_file_path: Path, mode: str = "r") -> Loom:
         abs_loom_file_path = self.get_loom_absolute_file_path(loom_file_path)
-        if not os.path.exists(abs_loom_file_path):
-            logger.error(f"The file {loom_file_path} does not exists.")
-            raise ValueError("The file located at " + abs_loom_file_path + " does not exist.")
-        # To check if the given file path is given specified url!
-        partial_md5_hash = LoomFileHandler.get_partial_md5_hash(abs_loom_file_path, 10000)
-        if abs_loom_file_path in self.active_looms:
-            logger.debug("Should be preloaded")
-            try:
-                logger.debug(
-                    f"Current mode: {self.active_looms[abs_loom_file_path].get_connection().mode}, wanted mode {mode}"
-                )
-
-                if self.active_looms[abs_loom_file_path].get_connection().mode == mode:
-                    loom = self.active_looms[abs_loom_file_path]
+        with self.file_locks[abs_loom_file_path]:
+            if not abs_loom_file_path.exists():
+                logger.error(f"The file {loom_file_path} does not exists.")
+                raise ValueError(f"The file located at {abs_loom_file_path} does not exist.")
+            if abs_loom_file_path in self.active_looms:
+                logger.debug("Should be preloaded")
+                loom = self.active_looms[abs_loom_file_path]
+                try:
                     logger.debug(
-                        f"Returning pre-loaded loom file {loom_file_path}. Hash {partial_md5_hash}, object {id(loom)}"
+                        f"Current mode: {self.active_looms[abs_loom_file_path].get_connection().mode}, wanted mode {mode}"
                     )
-                    return loom
-            except AttributeError:
-                logger.error("Loom was previously closed")
+                    if self.active_looms[abs_loom_file_path].get_connection().mode == mode:
+                        logger.debug(f"Returning pre-loaded loom file {loom_file_path}. Object {id(loom)}")
+                        return loom
+                    else:
+                        logger.error(
+                            f"Mode {mode} was requested for {loom_file_path}, but mode is currently {self.active_looms[abs_loom_file_path].get_connection().mode}"
+                        )
```

Comment:
Maybe add a `loom = None` here?

Reply:
This is something to decide on, _if_ `loom = None` works, it will result in the loom disappearing for the frontend user. Ideally, we actually want to respond with an error to let them know that it failed, but this is the bit w discussed about putting it in a different PR with the result class.